### PR TITLE
Add Claude Sonnet 3.5 VisionAgentCoder

### PR DESCRIPTION
**Pull request by dillonalaird**
Add support for Claude Sonnet 3.5 also fix some issues:
- Update extract_json to work on more side cases introduced with Claude Sonnet 3.5
- Redo debugger so instead of outputting code inside the JSON it outputs it inside markdown. This is because models, particularly Claude Sonnet 3.5, struggle to generate code inside JSON format. https://aider.chat/2024/08/14/code-in-json.html
- Fix bug in `extract_frames` documentation, was stating an incorrect default argument for FPS
- Allow debugger to see the tool documentation in case the bug was a missing import
- Add ability to print execution results without printing returned results. This is because all tool tracing gets added to the returned results which we don't want to show the model.
- Fixes edge case on OCR where it will fail if the image has 0 height or 0 width
- Updated plan reviewer prompt to better evaluate multiple plans

### CODE REVIEW

File: vision_agent/agent/agent_utils.py [hunk 1]
```diff
@@ -14,36 +14,48 @@ def _extract_sub_json(json_str: str) -> Optional[Dict[str, Any]]:
...
+    return json_str
+
+
+def _strip_markdown_code(inp_str: str) -> str:
+    pattern = r"```python.*?```"
+    cleaned_str = re.sub(pattern, "", inp_str, flags=re.DOTALL)
+    return cleaned_str
+
+
 def extract_json(json_str: str) -> Dict[str, Any]:
```
Comment (shankar-vision-eng) [inline vision_agent/agent/agent_utils.py:42]: Nice, looks much better

File: vision_agent/agent/vision_agent_coder.py [hunk 2]
```diff
@@ -451,17 +460,38 @@ def debug_code(
...
-                )
+                    feedback=format_memory(working_memory + new_working_memory),
+                ),
+                stream=False,
             )
+            fixed_code_and_test_str = cast(str, fixed_code_and_test_str)
+            fixed_code_and_test = extract_json(fixed_code_and_test_str)
+            code = extract_code(fixed_code_and_test_str)
+            if (
+                "which_code" in fixed_code_and_test
```
Comment (shankar-vision-eng) [inline vision_agent/agent/vision_agent_coder.py:484]: What is "which_code" here ?

File: vision_agent/agent/vision_agent_coder_prompts.py [hunk 3]
```diff
@@ -161,9 +174,10 @@
 
 **Instructions**:
 1. Given the plans, image, and tool outputs, decide which plan is the best to achieve the user request.
-2. Try solving the problem yourself given the image and pick the plan that matches your solution the best.
+2. Sovle the problem yourself given the image and pick the plan that matches your solution the best.
```
Comment (shankar-vision-eng) [inline vision_agent/agent/vision_agent_coder_prompts.py:177]: Typo "Solve"

File: vision_agent/agent/vision_agent_coder_prompts.py [hunk 4]
```diff
@@ -323,17 +342,24 @@ def find_text(image_path: str, text: str) -> str:
...
 {feedback}
 
-Please fix the bug by follow the error information and return a JSON object with the following format:
+Please fix the bug by correcting the error. Return the following JSON object followed by the fixed code in the below format:
+```json
 {{
     "reflections": str # any thoughts you have about the bug and how you fixed it
-    "code": str # the fixed code if any, else an empty string
-    "test": str # the fixed test code if any, else an empty string
+    "which_code": str # which code you fixed, can either be 'code' or 'test'
```
Comment (shankar-vision-eng) [inline vision_agent/agent/vision_agent_coder_prompts.py:356]: OK i can see what "which_code" means. I would rephrase this as 
`"which_code": str # The code block that was fixed, can only be 'code' or 'test'`

**Review by shankar-vision-eng — COMMENTED** [review]
Left few comments, will sync with you soon

File: vision_agent/agent/vision_agent_coder.py (hunk 2 above)
Comment (dillonalaird) [reply vision_agent/agent/vision_agent_coder.py:484]: I'm basically trying to get rid of it using JSON mode to output code because it leads to lower quality code (I added a link above)

**Review by shankar-vision-eng — CHANGES_REQUESTED** [review]
Can you add a test case for the ixc25_temporal_localization ?

**Review by shankar-vision-eng — APPROVED** [review]
LGTM
